File: 4.WhileLoop/app/src/main/java/com/example/martinrgb/a4whileloop/MainActivity.java
package com.example.martinrgb.a4whileloop;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        //WHILE LOOP

//        int[] elements = {5,7,12,100,8,-1,3};
//        int indexPos = 0;
//        int sumTotal = 0;
//
//        while(indexPos < elements.length)
//        {
//            System.out.println("Processing: " + indexPos + " Value: " + elements[indexPos]);
//
//            if(elements[indexPos]>99){
//
//                System.out.print("Continue!");
//                //跳过这一位
//                indexPos += 1;
//                //Forever 持续这个index的loop
//                continue;
//            }
//
//            if(elements[indexPos]<0){
//                //跳出whileloop
//                System.out.print("Break!");
//                break;
//            }
//
//
//            sumTotal = sumTotal + elements[indexPos];
//            indexPos++;
//        }
//
//        System.out.println("Sum total was:" + sumTotal);




        //FOR LOOP - 不设置索引号，拿数值本身来做循环

//        int[] elements = {5,7,12,100,8,-1,3};
//        int sumTotal = 0;
//
//       for(int elementContents : elements)
//        {
//            System.out.println("Processing: " + elementContents);
//
//            if(elementContents>99){
//
//                System.out.print("Continue!");
//                continue;
//            }
//
//            if(elementContents<0){
//                System.out.print("Break!");
//                break;
//            }
//
//
//            sumTotal = sumTotal + elementContents;
//        }
//
//        System.out.println("Sum total was:" + sumTotal);



        // Do While Loop

        int[] elements = {5,7,12,100,8,-1,3};
        int indexPos = 0;
        int sumTotal = 0;


        do{
            System.out.println("Processing: " + indexPos + " Value: " + elements[indexPos]);

            if(elements[indexPos]>99){

                System.out.print("Continue!");
                //跳过这一位
                indexPos += 1;
                //Forever 持续这个index的loop
                continue;
            }

            if(elements[indexPos]<0){
                //跳出whileloop
                System.out.print("Break!");
                break;
            }


            sumTotal = sumTotal + elements[indexPos];
            indexPos++;
        }while(indexPos < elements.length);

        System.out.println("Sum total was:" + sumTotal);




    }

}

File: 0.prototye_learning/56.Kotlin_HelloWorld/app/src/main/java/com/example/martinrgb/myapplication/MainActivity.kt
package com.example.martinrgb.myapplication

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import com.facebook.rebound.SimpleSpringListener
import com.facebook.rebound.Spring;
import com.facebook.rebound.SpringConfig;
import com.facebook.rebound.SpringListener;
import com.facebook.rebound.SpringSystem;
import com.facebook.rebound.SpringUtil;

class MainActivity : AppCompatActivity() {

    private var isTriggered = false
    private var mTextView: TextView? = null
    private val mconfig = SpringConfig.fromOrigamiTensionAndFriction(150.0, 10.0)
    private var mSpringSystem: SpringSystem? = null
    private var mSpring: Spring? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mTextView = findViewById(R.id.textView) as TextView
        setSpringSystem()
        findViewById(R.id.textView).setOnClickListener {
            isTriggered = !isTriggered
            if (isTriggered) {
                mSpring!!.setEndValue(1.0)
            } else {
                mSpring!!.setEndValue(0.0)
            }
        }
    }

    fun setSpringSystem() {
        mSpringSystem = SpringSystem.create()
        mSpring = mSpringSystem!!.createSpring()
        mSpring!!.springConfig = mconfig

        mSpring!!.addListener(object : SimpleSpringListener() {
            override fun onSpringUpdate(mSpring: Spring?) {
                val value = mSpring!!.currentValue.toFloat()
                val value1 = SpringUtil.mapValueFromRangeToRange(value.toDouble(), 0.0, 1.0, 1.0, 2.0).toFloat()
                mTextView!!.scaleX = value1
                mTextView!!.scaleY = value1
                Log.e("progress", java.lang.Float.toString(value))
            }
        })
    }


}
